src/compose/ComposeMenu.js

```diff
@@ -63,7 +63,7 @@ class ComposeMenu extends PureComponent<Props> {
     // https://github.com/react-native-community/react-native-image-picker/blob/master/docs/Reference.md
     error?: string | void | null | false,
     uri: string,
-    fileName: string,
+    fileName: ?string,
```

Comment:
This shouldn't have typechecked anyway – this isn't a supertype of `Response`!

Flow isn't catching this because of the `$FlowFixMe` up top: `ImagePicker` is defined as `any`. Changing the import to `import * as ImagePicker` (and removing the suppression comment) lets Flow see the definitions.

It seems that either import method works, at least on Android. Which is surprising, to say the least – but perhaps there's some React linkage magic going on. Does it work both ways on iOS?

Reply:
Hmm, I think I see what you mean -- using `import * as ImagePicker` reveals that the callback we're passing, `this.handleImagePickerReponse`, isn't accepted by Flow because of the types specified by react-native-image-picker (specifically, .launchCamera expects to pass a Response to that callback, which doesn't include fileName being null, and ours does). I remember being vaguely surprised that no Flow errors were introduced when I made this change, and that I didn't have to add any suppressions. I should probably have at least made a note of my surprise there. Turns out it's because of this very general suppression, as you point out.

I wonder if we can avoid doing such a general suppression. If you get a moment, would you like to go over this in person? I've poked around a bit here and I'm not sure how best to use a modified Response type and a modified type for ImagePicker.launchCamera without making a local copy of react-native-image-picker, which would complicate staying up to date with newer versions. (Maybe we'd just need to make that local copy anyway.)